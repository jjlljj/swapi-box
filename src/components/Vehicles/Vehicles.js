import React, { Component } from 'react';
import propTypes from 'prop-types';
import './Vehicles.css';
import Card from '../Card/Card';

class Vehicles extends Component {
   
  async componentDidMount() {
    if (!this.props.cards) this.props.fetchData()
  }

  renderCards() {
    const { addToFav, cards } = this.props
    return cards.map(card => {
            return (<Card card={card} addToFav={ addToFav }/>)
    })
  }

  render() {
    return (
      <div className="card-container"> 
        { 
          this.props.cards &&
          this.renderCards()
        }
      </div>
    );
  }
};

export default Vehicles;
