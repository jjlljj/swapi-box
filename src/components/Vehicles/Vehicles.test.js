/* eslint-disable */

import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Vehicles from './Vehicles';

describe('Vehicles', () => {

  let renderedComponent
  let mockFetchData
  const mockCards = [
    {
      Model: "Digger Crawler",
      Passengers: "30",
      "Vehicle Class": "wheeled",
      cardType: "vehicles",
      name: "Sand Crawler"
    },
    {
      Model: "T-16 skyhopper",  
      Passengers: "1",
      "Vehicle Class": "repulsorcraft",
      cardType: "vehicles",
      name: "T-16 skyhopper"
    }
  ]

  beforeEach(()=> {
    mockFetchData = jest.fn()
    renderedComponent = shallow(<Vehicles cards={mockCards} fetchData={ mockFetchData } />)
  })

  it('should match snapshot when passed mock cards', () => {

    expect(renderedComponent).toMatchSnapshot()
  })

  it('should match snapshot when passed only one card', () => {
   const mockCard = [
     {
        Model: "T-16 skyhopper",  
        Passengers: "1",
        "Vehicle Class": "repulsorcraft",
        cardType: "vehicles",
        name: "T-16 skyhopper"
      }
    ]

    renderedComponent = shallow(<Vehicles cards={mockCard} fetchData={mockFetchData} />)

    expect(renderedComponent).toMatchSnapshot()
  })

  it('should call fetchData func passed when no cards are passed', () => {

    expect(mockFetchData).not.toHaveBeenCalled()

    renderedComponent = shallow(<Vehicles cards={undefined} fetchData={mockFetchData}/>)

    expect(mockFetchData).toHaveBeenCalled()
  })

})
