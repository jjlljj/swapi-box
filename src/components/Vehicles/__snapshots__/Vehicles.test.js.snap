// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vehicles should match snapshot when passed mock cards 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Vehicles
    cards={
        Array [
            Object {
              "Model": "Digger Crawler",
              "Passengers": "30",
              "Vehicle Class": "wheeled",
              "cardType": "vehicles",
              "name": "Sand Crawler",
            },
            Object {
              "Model": "T-16 skyhopper",
              "Passengers": "1",
              "Vehicle Class": "repulsorcraft",
              "cardType": "vehicles",
              "name": "T-16 skyhopper",
            },
          ]
    }
    fetchData={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          card={
                    Object {
                              "Model": "Digger Crawler",
                              "Passengers": "30",
                              "Vehicle Class": "wheeled",
                              "cardType": "vehicles",
                              "name": "Sand Crawler",
                            }
          }
/>,
        <Card
          card={
                    Object {
                              "Model": "T-16 skyhopper",
                              "Passengers": "1",
                              "Vehicle Class": "repulsorcraft",
                              "cardType": "vehicles",
                              "name": "T-16 skyhopper",
                            }
          }
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "Sand Crawler",
        "nodeType": "function",
        "props": Object {
          "card": Object {
            "Model": "Digger Crawler",
            "Passengers": "30",
            "Vehicle Class": "wheeled",
            "cardType": "vehicles",
            "name": "Sand Crawler",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "T-16 skyhopper",
        "nodeType": "function",
        "props": Object {
          "card": Object {
            "Model": "T-16 skyhopper",
            "Passengers": "1",
            "Vehicle Class": "repulsorcraft",
            "cardType": "vehicles",
            "name": "T-16 skyhopper",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            card={
                        Object {
                                    "Model": "Digger Crawler",
                                    "Passengers": "30",
                                    "Vehicle Class": "wheeled",
                                    "cardType": "vehicles",
                                    "name": "Sand Crawler",
                                  }
            }
/>,
          <Card
            card={
                        Object {
                                    "Model": "T-16 skyhopper",
                                    "Passengers": "1",
                                    "Vehicle Class": "repulsorcraft",
                                    "cardType": "vehicles",
                                    "name": "T-16 skyhopper",
                                  }
            }
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Sand Crawler",
          "nodeType": "function",
          "props": Object {
            "card": Object {
              "Model": "Digger Crawler",
              "Passengers": "30",
              "Vehicle Class": "wheeled",
              "cardType": "vehicles",
              "name": "Sand Crawler",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "T-16 skyhopper",
          "nodeType": "function",
          "props": Object {
            "card": Object {
              "Model": "T-16 skyhopper",
              "Passengers": "1",
              "Vehicle Class": "repulsorcraft",
              "cardType": "vehicles",
              "name": "T-16 skyhopper",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Vehicles should match snapshot when passed only one card 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Vehicles
    cards={
        Array [
            Object {
              "Model": "T-16 skyhopper",
              "Passengers": "1",
              "Vehicle Class": "repulsorcraft",
              "cardType": "vehicles",
              "name": "T-16 skyhopper",
            },
          ]
    }
    fetchData={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          card={
                    Object {
                              "Model": "T-16 skyhopper",
                              "Passengers": "1",
                              "Vehicle Class": "repulsorcraft",
                              "cardType": "vehicles",
                              "name": "T-16 skyhopper",
                            }
          }
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "T-16 skyhopper",
        "nodeType": "function",
        "props": Object {
          "card": Object {
            "Model": "T-16 skyhopper",
            "Passengers": "1",
            "Vehicle Class": "repulsorcraft",
            "cardType": "vehicles",
            "name": "T-16 skyhopper",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            card={
                        Object {
                                    "Model": "T-16 skyhopper",
                                    "Passengers": "1",
                                    "Vehicle Class": "repulsorcraft",
                                    "cardType": "vehicles",
                                    "name": "T-16 skyhopper",
                                  }
            }
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "T-16 skyhopper",
          "nodeType": "function",
          "props": Object {
            "card": Object {
              "Model": "T-16 skyhopper",
              "Passengers": "1",
              "Vehicle Class": "repulsorcraft",
              "cardType": "vehicles",
              "name": "T-16 skyhopper",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
